airbyte-integrations/connectors/source-jdbc/src/main/java/io/airbyte/integrations/source/jdbc/AbstractJdbcSource.java

```diff
@@ -119,27 +122,52 @@ public AirbyteCatalog discover(JsonNode config) throws Exception {
     }
   }
 
+  protected void updateSchemaFocus(JsonNode config) {
+    if (config.has("schemaFocus")) {
+      this.schemaFocus = config.get("schemaFocus").asBoolean();
+    } else {
+      this.schemaFocus = false;
+    }
+  }
+
+  protected boolean getSchemaFocus() {
+    return this.schemaFocus;
+  }
+
+  protected List<String> getExcludedInternalSchemas() {
+    return List.of();
+  }
+
   protected List<TableInfo> getTables(final Database database) throws Exception {
     return discoverInternal(database).stream()
         .map(t -> {
           final List<Field> fields = Arrays.stream(t.fields())
               .map(f -> Field.of(f.getName(), jooqDataTypeToJsonSchemaType(f.getDataType())))
               .collect(Collectors.toList());
 
-          return new TableInfo(t.getName(), fields);
+          return new TableInfo(String.format("%s.%s", t.getSchema().getName(), t.getName()), fields);
         })
         .collect(Collectors.toList());
   }
 
   private List<Table<?>> discoverInternal(final Database database) throws Exception {
     return database.query(context -> {
-      final String databaseName = getCurrentDatabaseName(context);
-      final List<Schema> schemas = context.meta().getSchemas(databaseName);
-      if (schemas.size() > 1) {
-        throw new IllegalStateException("found multiple databases with the same name.");
+      if (getSchemaFocus()) {
```

Comment:
why would we ever want to call `discoverInternal` with schemaFocus set to false? if the read operation is using tables from schemas other than public, won't it also need to use the schemaFocus? it looks like in the current implementation it would only ever be set to true for the discover action.

Reply:
I was more thinking the schemaFocus would be false by default (ie discover tables from all schemas, other than system ones)

The only case we need to focus on a single schema at a time is actually in the tests where multiple schemas are created for each test...

in the spec.json / UI, it is not possible to specify a schema for the source today, this PR resolve that issue by making that choice in the catalog selection instead

Reply:
If i'm understanding what you're saying the idea is that because the table exists in the catalog the `read` command will be able to find it. i think that will break if you look at the implementation of the read command. It uses `discoverInternal` to verify the tables it is going to read from. If it can't find them, I think it will break. You could try a test to verify, but I think that's true. 

 It seems like this shouldn't be a configuration at all. It seems like we should always be looking at all of the schemas. I'm not understanding the case when we wouldn't want to. Can discuss more tomorrow AM if this isn't clear.

Reply:
Ok i removed the schemaFocus option and the previous behavior is not available anymore.

(i also adapted the test in MySql source to handle this change)